utreexo-p2p-bip.md

```diff
@@ -1,236 +1,360 @@
+It's still reasonable for a single node to download this extra data but little caching goes a long way in reducing the amount of data that one has to download.
+We define the new P2P messages for the inclusion proofs to support caching to reduce bandwidth while also allowing a high bandwidth, low-latency usage.
 
 # License
-This BIP is licensed under the BSD 3-clause license.
 
-# Requirements and Compatibility
+This BIP is licensed under the BSD-3-Clause license.
 
-Nodes implementing Utreexo can choose which messages to support.  There are a number of configurations possible, and this BIP does not restrict nodes to any subsets of messages.  That said, there are two likely types of nodes: Compact State Nodes (CSNs) which have the goal of minimizing data storage and download while performing block validation, and archive and bridge nodes which store more data and provide this data to CSNs.  Bridge nodes, nodes which can add inclusion proofs to mempool transactions, support the same set of messages as CSNs, and in fact should be indistinguishable from CSNs on the network.  Archive nodes can send messages such as block summaries and block proofs.
+# Overview
+
+## Requirements and Compatibility
+
+Nodes implementing Utreexo can choose which messages to support.
+There are a number of configurations possible, and this BIP does not restrict nodes to any subsets of messages.
+That said, there are three likely types of nodes: Compact State Nodes (CSNs) which have the goal of minimizing data storage and download while performing block validation, and archive and bridge nodes which store more data and provide this data to CSNs.
+Bridge nodes are nodes that can add inclusion proofs to mempool transactions, support the same set of messages as CSNs, and are in fact should be indistinguishable from CSNs on the network.
+Archive nodes are able to serve the blocks and the inclusion proofs. However, they are not able to generate the inclusion proofs as they do not keep the full UTXO set.
 
 Note that the archive and bridge capabilities of a node are separate; a bridge node can be bridge only, without previous block proof data, and an archive node doesn't need to be able to bridge.
 
-The one exception to this flexibility is that archive nodes must provide both block summaries and block proofs.   While theoretically possible to split these two resources, the bock summaries are quite small relative to the block proofs, and it simplifies clients to be able to rely on being able to request both over the same connection.
+The one exception to this flexibility is that archive nodes must provide both the blocks and the inclusion proofs.
+While theoretically possible to split these two resources, the bock summaries are quite small relative to the block proofs, and it simplifies clients to be able to rely on being able to request both over the same connection.
```

Comment:
The first paragraph says blocks, and the second says block summaries. I think the first one is wrong? (We should be able to serve only proofs and not blocks)

Reply:
Ah this is old. I'll get rid of the block summaries part